programs/x509/cert_write.c

```diff
@@ -207,6 +216,18 @@ struct options {
     int format;                 /* format                               */
 } opt;
 
+static void ip_string_to_bytes(const char *str, uint8_t *bytes, int maxBytes)
+{
+    for (int i = 0; i < maxBytes; i++) {
+        bytes[i] = (uint8_t) strtoul(str, NULL, 10);
+        str = strchr(str, '.');
+        if (str == NULL || *str == '\0') {
+            break;
+        }
+        str++;
+    }
+}
```

Comment:
This is too loose.  I recommend just using the libc functions until and Mbed TLS gets a proper internal parser.

Reply:
Created https://github.com/Mbed-TLS/mbedtls/issues/7519 to track it. It's on top of the priority list, along with removing support for leading zeroes in IPv4 parsing.